target/linux/ramips/dts/mt7621_hiwifi_hc5962.dts

```diff
@@ -44,6 +44,22 @@
 			linux,code = <KEY_RESTART>;
 		};
 	};
+
+	ubi-concat {
+		compatible = "mtd-concat";
+		devices = <&ubipart0 &ubipart1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			ubi@540000 {
+				label = "ubi";
+				reg = <0x0 0x79c0000>;
+			};
+		};
+	};
```

Comment:
I'm not sure if/how badblock handling works with mtd-concat. Did you spend time with this?

If not, I'm fine with enlarging the kernel partition but would like to keep this out of here.

Reply:
It will be handled by UBI.
cc @realmicu

Reply:
I've splitted the commit into two. The first one enlarges the kernel partition and the second one uses mtd-concat.

The mtd-concat idea was from ad6c6361f9f85f7cf6f53558d109c82de73743a5 by @realmicu

Reply:
I confirm, I've never had any problems with such mtd-concat setup on my Netgear WNDR4300 v1. 
Please be aware though that using all flash on some routers may reveal low quality of flash chip validation, something I was able to fix from u-boot prompt : https://forum.openwrt.org/t/support-for-netgear-wndr4300-v2/26099/24